packages/@aws-cdk/aws-ecr/lib/repository.ts

```diff
@@ -422,6 +430,36 @@ export class Repository extends RepositoryBase {
       resource: 'repository',
       resourceName: this.physicalName,
     });
+
+    // image scanOnPush
+    if (props.imageScanOnPush) {
+      new cr.AwsCustomResource(this, 'ImageScanOnPush', {
+        resourceType: 'Custom::ECRImageScanOnPush',
+        onUpdate: {
+          service: 'ECR',
+          action: 'putImageScanningConfiguration',
+          parameters: {
+            repositoryName: this.repositoryName,
+            imageScanningConfiguration: {
+              scanOnPush: props.imageScanOnPush,
+            },
+          },
+          physicalResourceId: cr.PhysicalResourceId.of(this.repositoryName),
+        },
+        onDelete: {
+          service: 'ECR',
+          action: 'putImageScanningConfiguration',
+          parameters: {
+            repositoryName: this.repositoryName,
+            imageScanningConfiguration: {
+              scanOnPush: false,
+            },
+          },
+          physicalResourceId: cr.PhysicalResourceId.of(this.repositoryName),
+        },
+        policy: cr.AwsCustomResourcePolicy.fromSdkCalls({ resources: cr.AwsCustomResourcePolicy.ANY_RESOURCE }),
```

Comment:
Shouldn't this be restricted to `this.repositoryArn`?

Reply:
You marked this as resolved but didn't change it?

Reply:
Hi @jogold 

I did.  Did I miss anything?

https://github.com/aws/aws-cdk/blob/18dec7845a697139c368f27db0b7b077e6c7a6e6/packages/%40aws-cdk/aws-ecr/lib/repository.ts#L447-L458